pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go

```diff
@@ -162,25 +165,25 @@ func (c *StaticPodStateController) sync(ctx context.Context, syncCtx factory.Syn
 	}
 
 	// update failing condition
-	cond := operatorv1.OperatorCondition{
-		Type:   condition.StaticPodsDegradedConditionType,
-		Status: operatorv1.ConditionFalse,
-	}
+	cond := applyoperatorv1.OperatorCondition().
+		WithType(condition.StaticPodsDegradedConditionType).
+		WithStatus(operatorv1.ConditionFalse)
 	// Failing errors
 	if failingErrorCount > 0 {
-		cond.Status = operatorv1.ConditionTrue
-		cond.Reason = "Error"
-		cond.Message = v1helpers.NewMultiLineAggregate(errs).Error()
+		cond.WithStatus(operatorv1.ConditionTrue).
```

Comment:
note that we don't have to reassign the updated cond since we are dealing with pointers.
we should fix the places that already have been changed for consistency.

Reply:
IMO reassigning the variable looks more natural with the builder pattern. I guess it boils down to personal preference (but I agree we should keep consistency).

Reply:
OK, given that in the other places we already reassign I will update this PR to do the same.

Reply:
Updated, PTAL.